backend/src/cmd/import_realm_tree.rs

```diff
@@ -62,31 +64,135 @@ pub(crate) async fn run(args: &Args, config: &Config) -> Result<()> {
         .context("failed to check/run DB migrations")?;
     let conn = db.get().await?;
 
-    let mut dummy_blocks = DummyBlocks::new(args.dummy_blocks, &**conn).await?;
-
+    if args.dummy_blocks {
+        add_dummy_blocks(&mut root, &**conn).await?;
+    }
+    
     info!("Starting to insert realms into the DB...");
-    insert_realm(&**conn, &root, 0, &mut dummy_blocks).await?;
+    insert_realm(&**conn, &root, 0).await?;
     info!("Done inserting realms");
 
     Ok(())
 }
 
+// Wrapper function for recursively adding different blocks.
+fn for_all_realms(realm: &mut Realm, mut f: impl FnMut(&mut Realm)) {
+    fn add_blocks(realm: &mut Realm, f: &mut impl FnMut(&mut Realm)) {
+        f(realm);
+
+        for child in &mut realm.children {
+            add_blocks(child, f);
+        }
+    }
+
+    add_blocks(realm, &mut f);
+}
+
+// This adds different kinds of blocks to empty realms.
+async fn add_dummy_blocks(root: &mut Realm, db: &impl GenericClient) -> Result<()> {
+    let mut rng = thread_rng();
+
+    // Get relevant series fields and store them in hashmap.
+    let series_rows = db.query("select title, opencast_id from series", &[]).await?;
+    let mut series = <HashMap<_, Vec<_>>>::new();
+
+    for row in series_rows {
+        let Some(title) = row.get::<_, Option<String>>("title") else {
+            continue;
+        };
+        let oc_id = row.get::<_, String>("opencast_id");
+        series.entry(title).or_default().push(oc_id);
+    }
+
+    // Vector containing all event ids.
+    let event_rows = db.query("select id from events", &[]).await?;
+    let events = event_rows.into_iter()
+        .map(|row| row.get::<_,i64>("id"))
+        .collect::<Vec<_>>();
+
+    // Recursive functions to populate realms and children realms:
+    // Insert matching series blocks.
+    for_all_realms(root, |realm| {
+        if realm.blocks.is_empty() {
```

Comment:
I just noticed that all usages of `for_all_realms` have this `if realm.blocks.is_empty()` at the start. I think we can just move that into `for_all_realms` and rename it to `for_all_empty_realms`. And adjust its doc comment appropriately.